modules/dnn/src/layers/batch_norm_layer.cpp

```diff
@@ -272,6 +277,15 @@ class BatchNormLayerImpl CV_FINAL : public BatchNormLayer
         inputs_arr.getMatVector(inputs);
         outputs_arr.getMatVector(outputs);
 
+        if (inputs[0].dims <= 1) { // Handling for 0D and 1D
+            Mat &inpBlob = inputs[0];
+            Mat &outBlob = outputs[0];
+            CV_Assert(inpBlob.total() == weights_.total());
+            cv::multiply(inpBlob, weights_, outBlob);
+            cv::add(outBlob, bias_, outBlob);
+            return;
+        }
```

Comment:
Does OpenCL backend supports 1D?

Reply:
I do not think it does. On CI, I can not find any information related to OCV. I did not change anything in `#HAVE_OPENCL` part